src/operators.jl

```diff
@@ -45,8 +45,8 @@ Base.:(+)(lhs::Parameter, rhs::GAEv{C}) where {C} = PAE{C}(copy(rhs),GAEp{C}(zer
 Base.:(-)(lhs::Parameter, rhs::GAEv{C}) where {C} = PAE{C}(-rhs,GAEp{C}(zero(C), lhs => 1.))
 
 # Parameter--ParametrizedAffExpr{C}
-Base.:(+)(lhs::Parameter, rhs::PAE{C}) where {C} = PAE{C}(copy(rhs.v),lhs+rhs.p)
-Base.:(-)(lhs::Parameter, rhs::PAE{C}) where {C} = PAE{C}(-rhs.v,lhs-rhs.p)
+Base.:(+)(lhs::Parameter, rhs::PAE{C}) where {C} = PAE{C}(copy(rhs.v),JuMP.add_to_expression!(rhs.p,lhs))
+Base.:(-)(lhs::Parameter, rhs::PAE{C}) where {C} = PAE{C}(-rhs.v,JuMP.add_to_expression!(-1*rhs.p,lhs))

```

Comment:
We have to be careful here, `JuMP.add_to_expression!` updates `expression` *in place*, but the basic operations cannot destroy the original input. This is certainly happening in `+` and might be happening in `-`. Thats the reason for the previous implementation.

Reply:
I understand the problem here. 

Then we need to implemented properly `+(::Parameter, ::GenericAffExpr{Float64,Parameter})` because in b6116a9 what I did was calling `JuMP.add_to_expression!` and it will create the same update in place.

Reply:
I implemented `+(::Parameter, ::GenericAffExpr{Float64,Parameter})` and `-(::Parameter, ::GenericAffExpr{Float64,Parameter})` in the same way at it is done un JuMP

https://github.com/JuliaOpt/JuMP.jl/blob/7426454d1271e78a088b7fce1cdf0374cf6d6dd1/src/operators.jl#L131

See ac244ba